wangle/service/ServerDispatcher.h

```diff
@@ -94,11 +95,21 @@ class MultiplexServerDispatcher : public HandlerAdapter<Req, Resp> {
       : service_(service) {}
 
   void read(Context* ctx, Req in) override {
-    (*service_)(std::move(in)).then([ctx](Resp resp) {
-      ctx->fireWrite(std::move(resp));
+    std::tuple<Context*, std::shared_ptr<PipelineBase> > guard{ctx, ctx->getPipelineShared()};
```

Comment:
Don't need this tuple. Just `=` capture `ctx`, and capture `pipe = ctx->getPipelineShared()` below? That gets rid of the `std::get` calls.